sympy/physics/units/quantities.py

```diff
@@ -88,6 +88,10 @@ def _eval_is_positive(self):
     def _eval_is_constant(self):
         return self.scale_factor.is_constant()
 
+    def _eval_Abs(self):
+        return self.__class__(self.name, self.dimension,
```

Comment:
`self.func` is almost always the same as `self.__class__` (and it should work just fine with inheritance). However, we require classes to rebuild using `self.func` instead of `self.__class__` as sometimes we want to store additional information. The technical reason for this is that sometimes we want to allow both `f` and `f(x)` to be usable in expressions, but SymPy also requires every expression to be an instance (specifically an instance of `Basic`). Take a look at `sympy.assumptions.assume.AppliedPredicate` for an example.